homeassistant/components/derivative/sensor.py

```diff
@@ -114,6 +123,25 @@ def calc_derivative(entity, old_state, new_state):
             ):
                 return
 
+            now = new_state.last_updated
+            self._state_list.append((now, new_state.state))
+
+            # Get indices of tuples that are older than `unit_time`
```

Comment:
Aren't we using the `time_window` in this loop?
```suggestion
            # Get indices of tuples that are older and outside of the `time_window`
```

Reply:
Fixed! Thanks for the suggestion 👍